scripts/prepdocslib/textsplitter.py

```diff
@@ -16,16 +18,92 @@ def split_pages(self, pages: List[Page]) -> Generator[SplitPage, None, None]:
             yield  # pragma: no cover - this is necessary for mypy to type check
 
 
+STANDARD_LINE_BREAKS = [",", ";", ":", " ", "(", ")", "[", "]", "{", "}", "\t", "\n"]
+CJK_LINE_BREAKS = [
+    "。",
+    "、",
+    "，",
+    "；",
+    "：",
+    "（",
+    "）",
+    "【",
+    "】",
+    "「",
+    "」",
+    "『",
+    "』",
+    "〔",
+    "〕",
+    "〈",
+    "〉",
+    "《",
+    "》",
+    "〖",
+    "〗",
+    "〘",
+    "〙",
+    "〚",
+    "〛",
+    "〝",
+    "〞",
+    "〟",
+    "〰",
+    "〾",
+    "〿",
+    "–",
+    "—",
+    "‘",
+    "’",
+    "‚",
+    "‛",
+    "“",
+    "”",
+    "„",
+    "‟",
+    "‹",
+    "›",
+    "‽",
+    "⁇",
+    "⁈",
+    "⁉",
+    "⸮",
+    "ⸯ",
+    "。",
+    "々",
+    "〻",
+    "〼",
+]
+cl100k = tiktoken.get_encoding("cl100k_base")
+
+
+def split_page_by_max_tokens(page_num: int, text: str, max_tokens: int) -> Generator[SplitPage, None, None]:
+    """
+    Splits page by tokens
```

Comment:
```suggestion
    Splits given text into SplitPage objects that are shorter than the max_tokens.
```

The current docstring is basically the function name, so I suggest making it more specific or removing it.